CDH/image-rs | promote error information printing

Xynnn007 opened the pull request:
This PR is mainly for move image-pull to cdh. It improves the log printing after CDH RPC call failure to keep the error information consistent with that when calling image-rs directly, and avoids a lot of useless debug trace information.

In addition, this PR includes an additional commit to abandon the Unknown Snapshot type. Please see the commit message for the specific reason.  cc @ChengyuZhu6

ChengyuZhu6 commented on image-rs/src/pull.rs:
```diff
@@ -183,16 +187,19 @@ impl<'a> PullClient<'a> {
                     &destination,
                 )
-                .await?;
+                .await
```
I think we also need to update the returned error results in the unit tests.

Xynnn007 replied:
Yes. Fixed.

Xynnn007 commented:
It's ready for review now!

fitzthum commented on image-rs/src/signature/mod.rs:
```diff
@@ -15,13 +15,48 @@ pub use policy::Policy;
+
+    #[cfg(feature = "signature-simple")]
+    #[error("Malwared simple signing sigstore config")]
+    MalwaredSimpleSigningSigstoreConfig,
```
again

fitzthum commented on image-rs/src/stream/unpack.rs:
```diff
@@ -25,18 +26,90 @@ use tokio::{fs, io::AsyncRead};
+
+#[derive(Error, Debug)]
+pub enum UnpackError {
+    #[error("Failed to delete existed broken layer when unpacking")]
```
nit: `existed` -> `existing`

fitzthum commented on image-rs/src/stream/unpack.rs:
```diff
@@ -25,18 +26,90 @@ use tokio::{fs, io::AsyncRead};
+        source: io::Error,
+    },
+
+    #[error("Failed to set ownerships for path: {path}")]
```
nit: 'ownerships` -> 'ownership`

Xynnn007 replied:
This is to align with the [function ](https://github.com/confidential-containers/guest-components/blob/main/image-rs/src/unpack.rs#L210-L248) where both UID and GID are both changed. This might be proper to name as `ownerships`?

fitzthum reviewed (commented):
Functionality looks good. Made a few suggestions about the wording since these errors will be exposed to users.

Xynnn007 commented on image-rs/src/auth/mod.rs:
```diff
@@ -7,9 +7,17 @@ pub mod auth_config;
+#[derive(Error, Debug)]
+pub enum AuthError {
+    #[error("Malwared registry auth file")]
+    MalwaredRegistryAuthFile,
```
thanks, fixed

Xynnn007 commented on image-rs/src/builder.rs:
```diff
@@ -3,24 +3,49 @@
+    },
+
+    #[error("Initialize layer store failed: {source}")]
+    InitializeLayerStore {
```
Let me add some verbs like `failed` for this

Xynnn007 commented:
@fitzthum Hi, I realized that [the PR](https://github.com/sigstore/sigstore-rs/pull/467) to help sigstore support `http_proxy` config  has been merged, so I also updated the sigstore version in this PR. The latest sigstore version supports configuring http proxy, so we can configure both http and https proxy ​​at the CDH configuration file level. 

To distinguish them, I changed the corresponding configuration items to our usual names, namely `http_proxy`, `https_proxy` and `no_proxy`. These new contents can all be configured in initdata, so I put them in this PR, ptal.
